### ISSUE TITLE
Update to jni-rs v0.20 for android

### ISSUE THREAD

**Issue by qdot** [deviceplug/btleplug#281]
**Feature Description**

Lots of new features came in, but this will require some reworking jni-utils-rs.

**Comment by atouchet** [comment]
jni 0.22 is out now.

**Comment by qdot** [comment]
Yup, I'm aware. This port is... extremely difficult at this point, since the change from 0.19 to 0.22 spans multiple API epochs at this point. So I'm saving it until I have time to concentrate on it. Good things is that, with the new embedded test system, I at least have a way to verify on real hardware.

**Comment by qdot** [comment]
https://github.com/deviceplug/btleplug/pull/462

**Comment by qdot** [comment]
Updating to 0.22 in 0.13.0
